packages/client/src/write.ts

```diff
@@ -124,18 +103,35 @@ const _ = (async () => {
       0, // fuses
       zeroHash,
     ],
-    address: resolverAddr,
     account: signer,
-    value,
   }
 
   try {
-    await client.simulateContract(calldata)
+    await client.readContract({
+      address: universalResolver as Hex,
+      abi: universalResolverResolveAbi,
+      functionName: 'resolve',
+      args: [
+        encodedName,
+        encodeFunctionData({
+          functionName: 'writeParams',
+          abi,
+          args: [encodedName, encodeFunctionData(calldata)],
+        }),
+      ],
+    })
```

Comment:
this is the main difference where the view call is made instead of a transaction to gather the required parameters for the offchain call